schemas/claim.schema.json

```diff
@@ -64,27 +60,93 @@
           "type": "string",
           "description": "The end time of the test."
         },
-        "CapturedTestOutput": {
+        "capturedTestOutput": {
           "type": "string",
           "description": "Ginkgo writer output during the test run."
         },
         "testID": {
           "description": "The test identifier",
-          "$ref": "#/definitions/identifier"       
+          "$ref": "#/definitions/identifier"
+        },
+        "categoryClassification": {
+          "description": "Category classification for the test",
+          "$ref": "#/definitions/categoryClassification"
+        },
+        "catalogInfo": {
+          "description": "Test detailed information from catalog",
+          "$ref": "#/definitions/catalogInfo"
         }
-        
       },
       "additionalProperties": false,
       "required": [
-        "testText",
         "failureLocation",
         "failureLineContent",
         "state",
         "failureReason",
         "duration",
         "startTime",
-        "CapturedTestOutput",
-        "testId"
+        "capturedTestOutput",
+        "testID",
+        "categoryClassification",
+        "catalogInfo"
+      ]
+    },
+    "categoryClassification": {
+      "$id": "#categoryClassification",
+      "description": "categoryClassification is the classification for a single test case.",
+      "properties": {
```

Comment:
Suggestion: instead of making a key for each workload type, it could also be two arrays.
```
"properties": {
  "mandatoryFor" : {
    "type": "array",
    "items": {
      "type": string,
      "enum": ["telco", "non-telco", "far-edge", "extended"]
    }
  }
  "optionalFor" : {
    "type": "array",
    "items": {
      "type": string,
       "enum": ["telco", "non-telco", "far-edge", "extended"]
    }
  }
}
```
This is a json example:
```
...
"categoryClassification": {
  "mandatoryFor" : ["telco", "far-edge"]
  "optionalFor": ["non-telco", "extended"]
}
...
```
I think the enum for the workload type (array item type) could also be extracted as custom type, to avoid repetition.

Reply:
I would prefer to use the key per workload type because it makes it easier to parse in the TNF code and in the HTML parser. It also makes it easier to test the validity of the table: we would need to make sure that all the scenarios are distributed in either mandatory or optional sections and present only once. I didn't know about enum, would be useful to restrict ["Mandatory", "Optional"], I can add it